openedx/core/lib/api/permissions.py

```diff
@@ -54,7 +54,8 @@ class IsUserInUrl(permissions.BasePermission):
     def has_permission(self, request, view):
         # Return a 404 instead of a 403 (Unauthorized). If one user is looking up
         # other users, do not let them deduce the existence of an account.
-        if request.user.username != request.parser_context.get('kwargs', {}).get('username', None):
+        url_username = request.parser_context.get('kwargs', {}).get('username', '')
+        if request.user.username.lower() != url_username.lower():
```

Comment:
I didn't realize that we are case insensitive wrt username. But I tried it, and it won't let me create an account where the username differs only by case. So this seems correct.